src/rewriter/basic_rewrite_rcons.cpp

```diff
@@ -72,28 +72,13 @@ bool BasicRewriteRCons::prove(
 bool BasicRewriteRCons::postProve(
     CDProof* cdp, Node a, Node b, theory::TheoryId tid, MethodId mid)
 {
-  if (tid == theory::THEORY_BV)
-  {
     Node eq = a.eqNode(b);
-#define POST_PROVE_CASE(name) \
-    if (tryRule(cdp, eq, ProofRule::name, {eq[0]})) \
-    { \
-      Trace("trewrite-rcons") << "Reconstruct " << eq << " (from " << tid << ", " \
-                              << mid << ")"  << std::endl; \
-      return true; \
-    } \
-    /* end of macro */
-
-    POST_PROVE_CASE(BV_UMULO_ELIMINATE)
-    POST_PROVE_CASE(BV_SMULO_ELIMINATE)
-    POST_PROVE_CASE(BV_FLATTEN_ASSOC_COMMUTE)
-    POST_PROVE_CASE(BV_FLATTEN_ASSOC_COMMUTE_NO_DUPLICATES)
-    POST_PROVE_CASE(BV_ADD_COMBINE_LIKE_TERMS)
-    POST_PROVE_CASE(BV_MULT_SIMPLIFY)
-    POST_PROVE_CASE(BV_SOLVE_EQ)
-    POST_PROVE_CASE(BV_BITWISE_EQ)
-    POST_PROVE_CASE(BV_BITWISE_SLICING)
-  }
+    if (theory::TheoryId::THEORY_BV == tid && tryRule(cdp, eq, ProofRule::THEORY_REWRITE, {eq[0]}))
```

Comment:
I don't think this will be specific to THEORY_BV, so remove the first condition.

Reply:
I was going to ask about this. The bv proof checker will fail if it is not `THEORY_BV`. For example, some rules like `SolveEq` do not check if their inputs are actually bitvectors in `::applies`, so there has to be some guard on the theory id.